### PR TITLE
fix millisecond parsing errors

### LINKED ISSUE #894: One second off with pre epoch dates
second test below fails. This was probably introduced in #752

```
public void testEpoch() {
    assertParsedRfc3339(
        "1970-01-01T00:00:00.000Z", SecondsAndNanos.ofSecondsAndNanos(0, 0));
  }

  public void testOneSecondBeforeEpoch() {
    assertParsedRfc3339(
        "1969-12-31T23:59:59.000Z", SecondsAndNanos.ofSecondsAndNanos(-1, 0));
  }
```

### PR DESCRIPTION
**Pull request by elharo**
Fixes #894 @suztomo @olavloite @codyoss

### CODE REVIEW

File: google-http-client/src/test/java/com/google/api/client/util/DateTimeTest.java [hunk 1]
```diff
@@ -225,8 +220,18 @@ public void testParseRfc3339ToSecondsAndNanos() {
...
+  
+  public void testEpoch() {
+    assertParsedRfc3339(
+        "1970-01-01T00:00:00.000Z", SecondsAndNanos.ofSecondsAndNanos(0, 0));
+  }
+
+  public void testOneSecondBeforeEpoch() {
+    assertParsedRfc3339(
+        "1969-12-31T23:59:59.000Z", SecondsAndNanos.ofSecondsAndNanos(-1, 0));
+  }
```
Comment (suztomo) [inline google-http-client/src/test/java/com/google/api/client/util/DateTimeTest.java]: Would you add the following test too?

```
public void testGregorianCalendarStartDate() {
    assertParsedRfc3339(
        "1582-10-15T01:23:45.123Z", SecondsAndNanos.ofSecondsAndNanos(-12219287774L, -877000000));
}
```
(not 100% sure about the value of nanosecond part to represent `-12219287774877` milliseconds)

As per Beam's PubsubClientTest, date 1582-10-15 is an important date when Gregorian Calendar was adopted.
https://github.com/apache/beam/blob/master/sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubClientTest.java#L144

File: google-http-client/src/test/java/com/google/api/client/util/DateTimeTest.java [hunk 2]
```diff
@@ -15,6 +15,7 @@
 package com.google.api.client.util;
 
 import com.google.api.client.util.DateTime.SecondsAndNanos;
+
```
Comment (suztomo) [inline google-http-client/src/test/java/com/google/api/client/util/DateTimeTest.java]: Nit?

Comment (elharo) [reply google-http-client/src/test/java/com/google/api/client/util/DateTimeTest.java]: done

File: google-http-client/src/test/java/com/google/api/client/util/DateTimeTest.java (hunk 1 above)
Comment (elharo) [reply google-http-client/src/test/java/com/google/api/client/util/DateTimeTest.java]: I can't verify that this value is correct, so I don't want to add it here.

Comment (suztomo) [reply google-http-client/src/test/java/com/google/api/client/util/DateTimeTest.java]: I will try to add this in a separate PR then.

**Pull request merged**